crud/src/main/java/com/redhat/lightblue/rest/crud/cmd/FindCommand.java

```diff
@@ -119,32 +119,40 @@ public void write(OutputStream os) throws IOException {
                         writer.write(chunkNode.toString());
                     }
                     writer.flush();
-                } finally {
+                } catch(Exception e) {
+                    metricCtx.markRequestException(e);
+                }
+                finally {
                     streamResponse.documentStream.close();
+                    metricCtx.endRequestMonitoring();
                 }
             }
         };
     }
 
     @Override
     public CallStatus run() {
+    	metricCtx = metrics.startStreamingEntityRequest("find", entity, version);
         LOGGER.debug("run: entity={}, version={}", entity, version);
         Error.reset();
         Error.push("rest");
         Error.push(getClass().getSimpleName());
         Error.push(entity);
+        Response r = null;
         try {
             FindRequest ireq;
             try {
                 ireq = getJsonTranslator().parse(FindRequest.class, JsonUtils.json(request));
             } catch (Exception e) {
+                metricCtx.markRequestException(e);
                 LOGGER.error("find:parse failure: {}", e);
                 return new CallStatus(Error.get(RestCrudConstants.ERR_REST_FIND, "Error during the parse of the request"));
             }
             LOGGER.debug("Find request:{}", ireq);
             try {
                 validateReq(ireq, entity, version);
             } catch (Exception e) {
+                metricCtx.markRequestException(e);
```

Comment:
Oh nevermind, looks like you already added improvements around the lightblue Error objects :-). Nice! So yeah, using the Error.get value to log the errors here like the others should be more consistent.